crates/solver/src/settlement_rater.rs

```diff
@@ -140,7 +150,34 @@ impl SettlementRating for SettlementRater {
         prices: &ExternalPrices,
         gas_price: GasPrice1559,
     ) -> Result<(Vec<RatedSolverSettlement>, Vec<SimulationWithError>)> {
-        let simulations = self.simulate_settlements(settlements, gas_price).await?;
+        // first simulate settlements without internalizations to make sure they pass
+        let simulations = self
+            .simulate_settlements(
+                settlements,
+                gas_price,
+                InternalizationStrategy::EncodeAllInteractions,
+            )
+            .await?;
+
+        // split simulations into succeeded and failed groups, then do the rating only for succeeded settlements
+        let (settlements, simulations_failed): (Vec<_>, Vec<_>) = simulations
+            .into_iter()
+            .partition_map(|simulation| match simulation.gas_estimate {
+                Ok(_) => Either::Left((
+                    simulation.simulation.solver,
+                    simulation.simulation.settlement,
+                )),
+                Err(_) => Either::Right(simulation),
+            });
+
+        // since rating is done with internalizations, repeat the simulations for previously succeeded simulations
+        let mut simulations = self
```

Comment:
Alternatively doing this could be moved into `self.simulate_settlements`. I don't know what is cleaner.